tasks/main.yml

```diff
@@ -17,26 +12,328 @@
       {{ ansible_facts['distribution'] ~ '_' ~
       ansible_facts['distribution_version'] }}.yml
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __mssql_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __mssql_vars_file is file
+
+- name: "Verify that the user accepts EULA variables"
+  assert:
+    that:
+      - mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula
+      - mssql_accept_microsoft_cli_utilities_for_sql_server_eula
+      - mssql_accept_microsoft_sql_server_2019_standard_eula
+    fail_msg:
+      - "You must accept EULA by setting the following variables to true:"
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+  register: __mssql_eula_evaluation
+
+- name: Configure the Microsoft SQL Server 2019 repository
+  yum_repository:
+    name: packages-microsoft-com-mssql-server-2019-gdr
+    description: Microsoft SQL Server 2019
+    baseurl: "{{ __mssql_server_repository }}"
+    gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+    gpgcheck: true
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Ensure that the Microsoft SQL Server package is installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __mssql_server_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-    mode: 0400
-  notify: template handler to restart services
+- name: Check if the errorlogfile setting has a custom value
+  command: /opt/mssql/bin/mssql-conf get filelocation errorlogfile
```

Comment:
I would suggest a `shell` for this:
```yaml
shell: |
  set -euo pipefail
  errorlog=$(/opt/mssql/bin/mssql-conf get filelocation errorlogfile 2>&1 | sed 's/^.*[: ][: ]*//') || :
  if [ -z "${errorlog}" ] || [ ! -f "${errorlog}" ]; then
    errorlog=/var/opt/mssql/log/errorlog
    if [ ! -f "${errorlog}" ]; then
      errorlog=""
    fi
  fi
  echo "${errorlog}"
register: __mssql_errorlogfile
```
then you can use it in `when` like this for a positive test:
`when: __mssql_errorlogfile | length > 0`
or like this for a negative test
`when: not __mssql_errorlogfile`
not sure why `when: __mssql_errorlogfile | bool` doesn't work - it should evaluate to `true` if `__mssql_errorlogfile` is set to a non-empty string, and `false` if `__mssql_errorlogfile` is set to an empty string - but it doesn't seem to be working for me - so safest to use `length > 0` for the positive test.